httpcli/http.go

```diff
@@ -79,8 +98,11 @@ func (c *Client) Do(m encoding.Marshaler) (encoding.Decoder, io.Closer, error) {
 	req.Header.Set("Content-Type", c.codec.MediaTypes[0])
 	req.Header.Set("Accept", c.codec.MediaTypes[1])
 
-	res, err := c.cli.Do(req)
+	res, err := c.do(req)
 	if err != nil {
+		if res != nil && res.Body != nil {
```

Comment:
There is nothing in the API docs for http.Client.Do that says that if err
!= nil then res will be nil.

https://golang.org/pkg/net/http/#Client.Do

Or, am I mis-reading?



On Fri, Nov 13, 2015 at 5:49 AM, Tomás Senart <notifications@github.com>
wrote:

> In httpcli/http.go
> <https://github.com/mesos/mesos-go/pull/209#discussion_r44770397>:
>
> >  	if err != nil {
> > +		if res != nil && res.Body != nil {
>
> I don't get this. When there is an error, res should be always nil. This
> invariant must be part of the API.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mesos/mesos-go/pull/209/files#r44770397>.
>